### ISSUE TITLE
Creating new store or campaign sometimes logout

### ISSUE THREAD

**Issue by bjarnef** [TeaCommerce/Tea-Commerce-for-Umbraco#45]
I have tested with Umbraco 7.4.1 / SQL CE and Tea Commerce 3.1.0

I think I have had this issue before and it seems to happen first time you create a new store or marketing campaign .. or when there isn't any stores or campaigns already created.
It seems to happen immediately after reloading the child nodes for "Stores" or "Marketing".

I get this error in the console.

```
)]}',
{"Message":"Authorization has been denied for this request."}
```

![image](https://cloud.githubusercontent.com/assets/2919859/13404213/ea4f70bc-df18-11e5-9a88-2247df4f3178.png)

![image](https://cloud.githubusercontent.com/assets/2919859/13404224/fac81660-df18-11e5-9a3c-deac38e1911a.png)

**Comment by runegronkjaer** [comment]
Hi Bjarne,

I believe Umbraco will do that if the get nodes fails somehow. We did have a problem with creating stores that would fail yesterday and fixed a bug. The fix have been uploaded to our. I don't know if you where unlucky enough to get the faulty version of the package.

I will try and make a new installation myself and check it out.

/Rune

**Comment by runegronkjaer** [comment]
Looks like at the moment before a store have been created this will happen, and after creating the first store you will have to reload the page. Seems to have something todo with some legacy Umbraco code we use for the tree.
We might have to update that.

**Comment by bjarnef** [comment]
Hi Rune

Okay, I just noticed it.. but seems only to happen when there isn't any child nodes (it seems GetNodes method doesn't get any id's?

Btw. the documentation for 3.1.0 : https://documentation.teacommerce.net/revision-history/
The bindingRedirect for Newtonsoft.Json should of course use v8 as new version, not v6 :)

/Bjarne

**Comment by runegronkjaer** [comment]
Fixed the revision history, thanks :)

**Comment by mattbrailsford** [comment]
This had just been fixed in f71fd0b and will be in 3.2.6 release
